crates/paimon/src/spec/schema_change.rs

```diff
@@ -0,0 +1,519 @@
+        column_move: ColumnMove,
+    ) -> Self {
+        SchemaChange::AddColumn {
+            field_name,
+            data_type,
+            description: Some(description),
+            column_move: Some(column_move),
+        }
+    }
+
+    /// impl the `rename_column`.
+    pub fn rename_column(field_name: String, new_name: String) -> Self {
+        SchemaChange::RenameColumn {
+            field_name,
+            new_name,
+        }
+    }
+
+    /// impl the `drop_column`.
+    pub fn drop_column(field_name: String) -> Self {
+        SchemaChange::DropColumn { field_name }
+    }
+
+    /// impl the `update_column_type`.
+    pub fn update_column_type(field_name: String, new_data_type: DataType) -> Self {
+        SchemaChange::UpdateColumnType {
+            field_name,
+            data_type: new_data_type,
+        }
+    }
+
+    /// impl the `update_column_position`.
+    pub fn update_column_nullability(field_name: String, new_nullability: bool) -> Self {
+        SchemaChange::UpdateColumnNullability {
+            field_name: vec![field_name],
+            nullable: new_nullability,
+        }
+    }
+
+    /// impl the `update_columns_nullability`.
+    pub fn update_columns_nullability(field_names: Vec<String>, new_nullability: bool) -> Self {
+        SchemaChange::UpdateColumnNullability {
+            field_name: field_names,
+            nullable: new_nullability,
+        }
+    }
+
+    /// impl the `update_column_comment`.
+    pub fn update_column_comment(field_name: String, comment: String) -> Self {
+        SchemaChange::UpdateColumnComment {
+            field_names: vec![field_name],
+            new_description: comment,
+        }
+    }
+
+    /// impl the `update_columns_comment`.
+    pub fn update_columns_comment(field_names: Vec<String>, comment: String) -> Self {
+        SchemaChange::UpdateColumnComment {
+            field_names,
+            new_description: comment,
+        }
+    }
+
+    /// impl the `update_column_position`.
+    pub fn update_column_position(column_move: ColumnMove) -> Self {
+        SchemaChange::UpdateColumnPosition { column_move }
+    }
+}
+
+/// The type of move.
+///
+/// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L412>
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+pub enum ColumnMoveType {
+    FIRST,
+    AFTER,
+}
+
+/// Represents a requested column move in a struct.
+///
+/// Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-core/src/main/java/org/apache/paimon/schema/SchemaChange.java#L410>
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ColumnMove {
+    field_name: String,
+    referenced_field_name: Option<String>,
+    #[serde(rename = "type")]
+    move_type: ColumnMoveType,
+}
+
+impl ColumnMove {
+    /// Create a new `Move`.
+    pub fn new(
```

Comment:
I suggest to add `new_first` and `new_after` to replace `new` / `first` / `after`.